RecoTracker/LST/plugins/alpaka/LSTModulesDevESProducer.cc

```diff
@@ -0,0 +1,54 @@
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/ESProducer.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/ModuleFactory.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/memory.h"
+
+#include "RecoTracker/LST/interface/LSTModulesDev.h"
+#include "RecoTracker/Record/interface/TrackerRecoGeometryRecord.h"
+
+// LST includes
+#include <SDL/ModuleMethods.h>
+#include <SDL/LST.h>
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+  
+class LSTModulesDevESProducer : public ESProducer {
+public:
+  LSTModulesDevESProducer(const edm::ParameterSet &iConfig);
+
+  static void fillDescriptions(edm::ConfigurationDescriptions &descriptions);
+
+  std::optional<SDL::modulesBuffer<alpaka_common::DevHost>> produce(const TrackerRecoGeometryRecord &iRecord);
+
+private:
+  const std::string txtFile_;
+};
+
+LSTModulesDevESProducer::LSTModulesDevESProducer(const edm::ParameterSet &iConfig)
+  : ESProducer(iConfig), txtFile_{iConfig.getParameter<edm::FileInPath>("txt").fullPath()}
+{
+  setWhatProduced(this, iConfig.getParameter<std::string>("ComponentName"));
```

Comment:
The `ComponentName` is strictly speaking redundant, because the framework already provides the same functionality out of the box with `appendToDataLabel` parameter.